frontend/dockerfile/dockerfile2llb/convert_runmount.go

```diff
@@ -15,33 +15,36 @@ import (
 	"github.com/pkg/errors"
 )
 
-func detectRunMount(cmd *command, allDispatchStates *dispatchStates) bool {
-	if c, ok := cmd.Command.(*instructions.RunCommand); ok {
-		mounts := instructions.GetMounts(c)
-		sources := make([]*dispatchState, len(mounts))
-		for i, mount := range mounts {
-			if mount.From == "" && mount.Type == instructions.MountTypeCache {
-				mount.From = emptyImageName
-			}
-			from := mount.From
-			if from == "" || mount.Type == instructions.MountTypeTmpfs {
-				continue
-			}
-			stn, ok := allDispatchStates.findStateByName(from)
-			if !ok {
-				stn = &dispatchState{
-					stage:        instructions.Stage{BaseName: from},
-					deps:         make(map[*dispatchState]struct{}),
-					unregistered: true,
-				}
+func detectRunMount(cmd *command, allDispatchStates *dispatchStates) (bool, error) {
+	c, ok := cmd.Command.(*instructions.RunCommand)
+	if !ok {
+		return false, nil
+	}
+	mounts := instructions.GetMounts(c)
+	sources := make([]*dispatchState, len(mounts))
+	for i, mount := range mounts {
+		var from string
+		if mount.From == "" {
+			// this might not be accurate because the type might not have a real source (tmpfs for instance),
```

Comment:
I don't understand why this doesn't check for type anymore.

Reply:
If we want to be able to expand `type`, we can't check for it here. Fortunately (AFAICT), this function is responsible for getting mount sources so they can be registered, so we can use a fallback value (`emptyImageName` in this case) as a source, which again, AFAICT should be fine. Do you see any problem with that?